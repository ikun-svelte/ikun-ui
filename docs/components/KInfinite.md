---
title: KInfinite
lang: en-US
---

# KInfinite

Load more data while reach bottom of the page.

## Install

::: code-group

```bash [pnpm]
pnpm add @ikun-ui/infinite
```

```bash [yarn]
yarn add @ikun-ui/infinite
```

```bash [npm]
npm install @ikun-ui/infinite
```

:::

## Basic usage

<demo src="infinite/basic.svelte"  github='Infinite'></demo>

## Infinite Props

| Name          | Type                     | Default | Description                                |
| ------------- | ------------------------ | ------- | ------------------------------------------ |
| threshold     | `string`                 | `0`     | Threshold to call loadMore.                |
| elementScroll | `HTMLElement`            | `-`     | Element to bind scroll.                    |
| window        | `boolean`                | `false` | Bind scroll in window.                     |
| hasMore       | `boolean`                | `true`  | Tells you if there are more items to load. |
| horizontal    | `boolean`                | `false` | Changing orientation.                      |
| reverse       | `boolean`                | `false` | Revese scroll direction                    |
| cls           | `string`                 | `''`    | Additional class for                       |
| attrs         | `Record<string, string>` | `{}`    | Additional attributes                      |

## Infinite Events

| Name     | Description                                         | Type                          |
| -------- | --------------------------------------------------- | ----------------------------- |
| loadMore | Triggered when the scroll bar slides to the bottom. | `(event: CustomEvent)=> void` |

## Infinite Slots

| Name    | Description               |
| ------- | ------------------------- |
| default | Customize default content |
