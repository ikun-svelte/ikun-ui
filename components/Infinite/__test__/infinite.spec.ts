import { afterEach, expect, test, describe, beforeEach } from 'vitest';
import KInfinite from '../src';

let host: HTMLElement;

const initHost = () => {
	host = document.createElement('div');
	host.setAttribute('id', 'host');
	document.body.appendChild(host);
};
beforeEach(() => {
	initHost();
});
afterEach(() => {
	host.remove();
});

describe('Test: KInfinite', () => {
	test('props: cls', async () => {
		const instance = new KInfinite({
			target: host,
			props: {
				cls: 'k-infinite--test'
			}
		});
		expect(instance).toBeTruthy();
		expect((host as HTMLElement)!.innerHTML.includes('k-infinite--test')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});
});
