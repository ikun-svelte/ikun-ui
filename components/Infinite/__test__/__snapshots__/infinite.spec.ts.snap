// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test: KInfinite > props: cls 1`] = `"<div class=\\"k-infinite k-infinite--base k-infinite--test\\"> <div id=\\"svelte-infinite-scroll\\" style=\\"width: 0px;\\"></div></div>"`;
